spec/services/dqt_api_spec.rb

```diff
@@ -84,4 +84,87 @@
       end
     end
   end
+
+  describe ".unlock_teacher!" do
+    before do
+      FeatureFlag.activate(:unlock_teachers_self_service_portal_account)
+    end
+    after do
+      FeatureFlag.deactivate(:unlock_teachers_self_service_portal_account)
+    end
+
+    subject(:unlock_teacher!) do
+      described_class.unlock_teacher!(teacher_account)
+    end
+    let(:teacher_account) do
+      {
+        trn: "2921020",
+        emailAddresses: ["anonymous@anonymousdomain.org.net.co.uk"],
+        firstName: "Kevin",
+        lastName: "Evans",
+        dateOfBirth: "1990-01-01",
+        nationalInsuranceNumber: "AA123456A",
+        uid: "f7891223-7661-e411-8047-005056822391"
+      }
+    end
+
+    context "when teacher ID is found", vcr: true do
+      it { is_expected.to be_nil }
+    end
+
+    context "when teacher ID is not found", vcr: true do
+      let(:teacher_account) do
+        {
+          trn: "2921020",
+          emailAddresses: ["anonymous@anonymousdomain.org.net.co.uk"],
+          firstName: "Kevin",
+          lastName: "Evans",
+          dateOfBirth: "1990-01-01",
+          nationalInsuranceNumber: "AA123456A",
+          uid: "f6891223-7661-e431-8047-005056822391"
+        }
+      end
```

Comment:
If only the `uid` is different, can that be a `let(:uid)` and leave us with only one `let(:teacher_account)` hoisted at the top level that references the `uid`?

Also, can we make the non-matching UID look more obviously wrong? Like `a-non-matching-uid`.